Vagrantfile

```diff
@@ -0,0 +1,118 @@
+
+  # Create a public network, which generally matched to bridged network.
+  # Bridged networks make the machine appear as another physical device on
+  # your network.
+  # config.vm.network "public_network"
+
+  # Share an additional folder to the guest VM. The first argument is
+  # the path on the host to the actual folder. The second argument is
+  # the path on the guest to mount the folder. And the optional third
+  # argument is a set of non-required options.
+  # config.vm.synced_folder "../data", "/vagrant_data"
+
+  # Provider-specific configuration so you can fine-tune various
+  # backing providers for Vagrant. These expose provider-specific options.
+  # Example for VirtualBox:
+  #
+  # config.vm.provider "virtualbox" do |vb|
+  #   # Display the VirtualBox GUI when booting the machine
+  #   vb.gui = true
+  #
+  #   # Customize the amount of memory on the VM:
+  #   vb.memory = "1024"
+  # end
+  #
+  # View the documentation for the provider you are using for more
+  # information on available options.
+
+  # Enable provisioning with a shell script. Additional provisioners such as
+  # Ansible, Chef, Docker, Puppet and Salt are also available. Please see the
+  # documentation for more information about their specific syntax and use.
+  # config.vm.provision "shell", inline: <<-SHELL
+  #   apt-get update
+  #   apt-get install -y apache2
+  # SHELL
+  config.vm.provision "shell", privileged: false, inline: <<-SHELL
+	sudo apt-get update
+	
+	# Install pyenv prerequisites
+	sudo apt-get install -y build-essential libssl1.0-dev zlib1g-dev libbz2-dev libreadline-dev libsqlite3-dev wget curl llvm libncurses5-dev libncursesw5-dev xz-utils tk-dev libffi-dev liblzma-dev python-openssl git
+	
+	# Install pyenv
+	git clone https://github.com/pyenv/pyenv.git ~/.pyenv
+	
+	# Set the profile and the env if this is not done before
+	if [ ! $PYENV_ROOT ]
+	then
+		echo 'export PYENV_ROOT="$HOME/.pyenv"' >> ~/.profile
+		echo 'export PATH="$PYENV_ROOT/bin:$PATH"' >> ~/.profile
+		echo -e 'if command -v pyenv 1>/dev/null 2>&1; then\n  eval "$(pyenv init -)"\nfi' >> ~/.profile
+		source ~/.profile
+	fi
+	
+	# Install python 3.9.1
+	pyenv install 3.9.1
+	pyenv global 3.9.1
+	
+	# Install poetry
+	curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python
+	source ~/.profile
+	
+	# Install gunicorn
+	cd /vagrant
+	poetry run pip install gunicorn
+	
+  SHELL
+  
+  config.trigger.after :up do |trigger|
+	trigger.name = "Launching App"
+	trigger.info = "Running the TODO app setup script"
+	trigger.run_remote = {privileged: false, inline: "
+	# Install dependencies and launch
+	cd /vagrant
+	poetry install
+
+	# uncomment the below if you want to run flask instead of gunicorn
+	# poetry run flask run --host 0.0.0.0
+	
+	# Run gunicorn
+	poetry run gunicorn -b 0.0.0.0:5000 'todo_app.app:app' --daemon --access-logfile gunicorn-access.log --log-file gunicorn.log
```

Comment:
This section should be indented another level.

Reply:
Done